### PR TITLE
OSDOCS-4092:Add the common terms used in storage book

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.8+

Issue: https://issues.redhat.com/browse/OSDOCS-4092

Link to docs preview: https://50153--docspreview.netlify.app/openshift-enterprise/latest/storage/index.html#openshift-storage-common-terms_storage-overview

Additional information:

### CODE REVIEW

File: modules/openshift-storage-common-terms.adoc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Container Storage Interface (CSI)::
+An API specification for the management of container storage across different container orchestration (CO) systems.
+
+Dynamic Provisioning::
+The framework allows you to create storage volumes on-demand, eliminating the need for cluster administrators to pre-provision persistent storage.
+
+Ephemeral storage::
+Pods and containers can require temporary or transient local storage for their operation. The lifetime of this ephemeral storage does not extend beyond the life of the individual pod, and this ephemeral storage cannot be shared across pods.
+
+External CSI Controllers::
```
Comment (duanwei33) [inline modules/openshift-storage-common-terms.adoc:31]: I think we don't need this item, and the explanation is not accurate.

File: modules/openshift-storage-common-terms.adoc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+A deployment that deploys one or more pods with five containers.
+
+Fibre channel:: A networking technology that is used to transfer data among data centers, computer servers, switches and storage.
```
Comment (duanwei33) [inline modules/openshift-storage-common-terms.adoc:34]: Fibre -> Fiber

File: modules/openshift-storage-common-terms.adoc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+A pod is the smallest compute unit defined, deployed, and managed.
+
+Reclaim policy::
+A policy that tells the cluster what to do with the volume after it is released. A volume’s reclaim policy can be `Retain`, `Recycle`, or `Delete`.
+
+Role-based access control (RBAC):: Role-based access control (RBAC) is a method of regulating access to computer or network resources based on the roles of individual users within your organization.
+
+Stateless applications:: A stateless application is an application program that does not save client data generated in one session for use in the next session with that client.
+
+Stateful applications:: A stateful application is an application program that saves data to persistent disk storage. A server, client, and applications can use a persistent disk storage. Stateful applications manage the deployment and scaling of a set of Pods, and provides guarantee about the ordering and uniqueness of these Pods.
```
Comment (duanwei33) [inline modules/openshift-storage-common-terms.adoc:73]:
```suggestion
Stateful applications:: A stateful application is an application program that saves data to persistent disk storage. A server, client, and applications can use a persistent disk storage. Statefulset in {product-title} is used to manage the deployment and scaling of a set of Pods, and provides guarantee about the ordering and uniqueness of these Pods.
```

@xenolinux sorry, I need to add few words here compared to draft version.

File: modules/openshift-storage-common-terms.adoc (hunk 1 above)
Comment (xenolinux) [reply modules/openshift-storage-common-terms.adoc]: Addressed

File: modules/openshift-storage-common-terms.adoc (hunk 2 above)
Comment (xenolinux) [reply modules/openshift-storage-common-terms.adoc]: Addressed

File: modules/openshift-storage-common-terms.adoc (hunk 3 above)
Comment (xenolinux) [reply modules/openshift-storage-common-terms.adoc:70]: No worries :)

Addressed

File: modules/openshift-storage-common-terms.adoc [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Access modes:: Volume access modes describe volume capabilities. You can use access modes to match persistent volume claim (PVC) and persistent volume (PV). The following are the examples of access modes:
+
+* ReadWriteOnce (RWO)
+* ReadOnlyMany (ROX)
+* ReadWriteMany (RWX)
+* ReadWriteOncePod (RWOP)
+
+Cinder:: The Block Storage service for {rh-openstack-first} which manages the administration, security, and scheduling of all volumes.
+
+Config map:: A config map provides a way to inject configuration data into pods. You can reference the data stored in a config map in a volume of type `ConfigMap`. Applications running in a pod can consume this data.
```
Comment (tmalove) [inline modules/openshift-storage-common-terms.adoc:20]:
```suggestion
Config map:: A config map provides a way to inject configuration data into pods. You can reference the data stored in a config map in a volume of type `ConfigMap`. Applications running in a pod can use this data.
```
Suggestion only per the ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage

File: modules/openshift-storage-common-terms.adoc [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+Dynamic Provisioning::
+The framework allows you to create storage volumes on-demand, eliminating the need for cluster administrators to pre-provision persistent storage.
+
+Ephemeral storage::
+Pods and containers can require temporary or transient local storage for their operation. The lifetime of this ephemeral storage does not extend beyond the life of the individual pod, and this ephemeral storage cannot be shared across pods.
+
+Fiber channel:: A networking technology that is used to transfer data among data centers, computer servers, switches and storage.
+
+FlexVolume:: FlexVolume is an out-of-tree plugin interface that uses an exec-based model to interface with storage drivers. You must install the FlexVolume driver binaries in a pre-defined volume plugin path on each node and in some cases the control plane nodes.
```
Comment (tmalove) [inline modules/openshift-storage-common-terms.adoc:33]:
```suggestion
FlexVolume:: FlexVolume is an out-of-tree plug-in interface that uses an exec-based model to interface with storage drivers. You must install the FlexVolume driver binaries in a pre-defined volume plug-in path on each node and in some cases the control plane nodes.
```

File: modules/openshift-storage-common-terms.adoc (hunk 4 above)
Comment (xenolinux) [reply modules/openshift-storage-common-terms.adoc:20]: acked

File: modules/openshift-storage-common-terms.adoc (hunk 5 above)
Comment (xenolinux) [reply modules/openshift-storage-common-terms.adoc]: acked

**Comment by jeana-redhat** [comment]
version in comment 0 says 4.6+, I assume because this was opened before we closed 4.6-4.7. Will CP to 4.8+ unless I hear otherwise

**Comment by xenolinux** [comment]
> version in comment 0 says 4.6+, I assume because this was opened before we closed 4.6-4.7. Will CP to 4.8+ unless I hear otherwise

@jeana-redhat Yes correct. This should be cherry-picked to 4.8+. This was opened before we closed 4.6-7

**Pull request merged**